### PR TITLE
[CARBONDATA-728] add intergation with presto

### PR DESCRIPTION
**Pull request by ffpeng90**
add raw version of presto intergation 
---

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by chenliang613** [comment]
add to whitelist

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1100/

File: integration/presto/pom.xml [hunk 1]
```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.facebook.presto</groupId>
```
Comment (chenliang613) [inline integration/presto/pom.xml]: please change groupId to org.apache.carbondata. 
you can take integration/spark module as reference, and update accordingly.

**Comment by jackylk** [comment]
Thanks for working on this. Can you describe what feature is added in term of:
1. What SQL syntax is supported? DDL &DML?
2. I think it uses CarbonInputFormat to read, so are you pushing down column projection and filtering by setting the configuration in CarbonInputFormat?
3. Is there any SQL optimization integration with Presto's optimizer? like leveraging carbon's global dictionary to do lazy decode?

**Comment by jackylk** [comment]
And I think it is easier to review and can be merged sooner if you could break this PR down into smaller one. Just provide the very basic functionality in the first round of the integration. You can add more functionality in subsequent PR.

**Comment by ffpeng90** [comment]
Hi:
    1. This version only suppport DML,  
        All tables for test are created by spark-sql(DML part), 
        and i submit queries to presto to get results.
        I only tested the "Select" Case , like where, group , sum , join.


    2.  I use APIs like createQueryPlan, resolveFilter from class "CarbonInputFormatUtil". 
       To read carbon formatted table, i make the read process into several steps:
       a). load table metadata 
       b). get splits from table (pushing down filtering to filter datablocks of one segment @CarbonTableReader.getInputSplits2 )
       c). parse records ( pushing down column projection and filtering into QueryModel  @CarbondataRecordSetProvider.getRecordSet ) 


    3. As i described  in partC "parse records", I use QueryModel to get  decoded records.
       For lazy decoding,  I will keep on exploring a better solution.  Maybe we can get inspiration from module presto-orc, presto-parquet.
      
 
      
   






At 2017-03-15 09:11:19, "Jacky Li" <notifications@github.com> wrote:


Thanks for working on this. Can you describe what feature is added in term of:

What SQL syntax is supported? DDL &DML?
I think it uses CarbonInputFormat to read, so are you pushing down column projection and filtering by setting the configuration in CarbonInputFormat?
Is there any SQL optimization integration with Presto's optimizer? like leveraging carbon's global dictionary to do lazy decode?

**Comment by ffpeng90** [comment]
I'm focusing on two things.
1. let user can debug presto-carbondata in his IDE.
2. use new presto API to support lazy decode.
They will be ok soon.






At 2017-03-15 10:52:01, "彭" <ffp900510@126.com> wrote:

Hi:
    1. This version only suppport DML,  
        All tables for test are created by spark-sql(DML part), 
        and i submit queries to presto to get results.
        I only tested the "Select" Case , like where, group , sum , join.


    2.  I use APIs like createQueryPlan, resolveFilter from class "CarbonInputFormatUtil". 
       To read carbon formatted table, i make the read process into several steps:
       a). load table metadata 
       b). get splits from table (pushing down filtering to filter datablocks of one segment @CarbonTableReader.getInputSplits2 )
       c). parse records ( pushing down column projection and filtering into QueryModel  @CarbondataRecordSetProvider.getRecordSet ) 


    3. As i described  in partC "parse records", I use QueryModel to get  decoded records.
       For lazy decoding,  I will keep on exploring a better solution.  Maybe we can get inspiration from module presto-orc, presto-parquet.
      
 
      
   






At 2017-03-15 09:11:19, "Jacky Li" <notifications@github.com> wrote:


Thanks for working on this. Can you describe what feature is added in term of:

What SQL syntax is supported? DDL &DML?
I think it uses CarbonInputFormat to read, so are you pushing down column projection and filtering by setting the configuration in CarbonInputFormat?
Is there any SQL optimization integration with Presto's optimizer? like leveraging carbon's global dictionary to do lazy decode?

**Comment by chenliang613** [comment]
@ffpeng90  Thank you working on this feature.
1. Please remove author info
2.Please change chinese comment info to english
3.Please remove presto module from parent/pom.xml to let CI run OK.

**Comment by chenliang613** [comment]
@ffpeng90  please update the PR title also.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1290/

**Comment by ffpeng90** [comment]
as your wish

**Comment by chenliang613** [comment]
retest this please

**Comment by chenliang613** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1296/

**Comment by chenliang613** [comment]
@ffpeng90 
In presto/pom.xml, please change groupid from "com.facebook.presto" to "org.apache.carbondata"

**Pull request closed without merging**